lua/neogit/buffers/commit_editor/init.lua

```diff
@@ -40,25 +39,24 @@ function M:open()
     modifiable = true,
     readonly = false,
     autocmds = {
-      ["BufWritePre"] = function()
-        written = true
-      end,
       ["BufUnload"] = function(o)
-        if written then
-          if
-            not config.values.disable_commit_confirmation
-            and not input.get_confirmation("Are you sure you want to commit?")
-          then
-            -- Clear the buffer, without filling the register
-            vim.api.nvim_buf_set_lines(o.buf, 0, -1, false, {})
-            vim.api.nvim_buf_call(o.buf, function()
-              vim.cmd("silent w!")
-            end)
-          end
+        if vim.bo[o.buf].mod then
+          return
+        end
+
+        if
+          not config.values.disable_commit_confirmation
+          and not input.get_confirmation("Are you sure you want to commit?")
+        then
+          -- Clear the buffer, without filling the register
+          vim.api.nvim_buf_set_lines(o.buf, 0, -1, false, {})
+          vim.api.nvim_buf_call(o.buf, function()
+            vim.cmd("silent w!")
+          end)
         end
 
         if self.on_unload then
-          self.on_unload(written)
```

Comment:
Can't be anything else then `true` at this point. As we check for `written` above.
Going through the code base it's unclear to me why the bool value is passed here at all.  If you have the change to clarify maybe it gives the insight for further improvements.